File: libs/gomodels/channel_command_group.go
package model

type ChannelCommandGroup struct {
	ID        string `gorm:"primary_key;AUTO_INCREMENT;column:id;type:UUID;"  json:"id"`
	ChannelID string `gorm:"column:channelId;type:TEXT;"  json:"channelId"`
	Name      string `gorm:"column:name;type:TEXT;"  json:"name"`
	Color     string `gorm:"column:color;type:TEXT;"  json:"color"`

	Commands []ChannelsCommands `gorm:"foreignKey:GroupID" json:"commands"`
}

func (c *ChannelCommandGroup) TableName() string {
	return "channels_commands_groups"
}

File: libs/typeorm/src/entities/ChannelCommandGroup.ts
import { Column, Entity, JoinColumn, ManyToOne, OneToMany, PrimaryGeneratedColumn } from 'typeorm';

// eslint-disable-next-line import/no-cycle
import { Channel } from './Channel';
// eslint-disable-next-line import/no-cycle
import { ChannelCommand } from './ChannelCommand';

@Entity({ name: 'channels_commands_groups' })
export class ChannelCommandGroup {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  channelId: string;

  @ManyToOne(() => Channel, _ => _.commandsGroups)
  @JoinColumn({ name: 'channelId' })
  channel?: Channel;

  @Column()
  name: string;

  @OneToMany(() => ChannelCommand, _ => _.group)
  commands?: ChannelCommand[];

  @Column({ default: 'rgba(37, 38, 43, 1)' })
  color: string;
}